gateway/dataimpl/server_v1/routingserver.go

```diff
@@ -0,0 +1,109 @@
+package server_v1
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/couchbase/gocbcorex/cbmgmtx"
+	"github.com/couchbase/gocbcorex/contrib/cbconfig"
+	"github.com/couchbase/goprotostellar/genproto/routing_v2"
+	"go.uber.org/zap"
+)
+
+type RoutingServer struct {
+	routing_v2.UnimplementedRoutingServiceServer
+
+	logger       *zap.Logger
+	errorHandler *ErrorHandler
+	authHandler  *AuthHandler
+
+	bootstrapNodeIsLocal bool
+
+	// The hostname of the node that CNG was bootstrapped agaisnt. We can't just
+	// use thisNode because that just depends on the node we happen to get the
+	// bucket config from
+	bootstrapHost string
```

Comment:
A future improvement that's likely worth doing is to cover the alternate address case and cluster_run cases by using the "canonical id" for a node here (its primary hostname + its ns_server http port).